inabox/tests/setup_disperser_harness.go

```diff
@@ -392,3 +422,145 @@ func stopAllRelays(servers []*relay.Server, logger logging.Logger) {
+	}
+
+	// Create controller logger config for file output
+	loggerConfig := common.LoggerConfig{
+		Format:       common.TextLogFormat,
+		OutputWriter: io.MultiWriter(os.Stdout, logFile),
+		HandlerOpts: logging.SLoggerOptions{
+			Level:     slog.LevelDebug,
+			NoColor:   true,
+			AddSource: true,
+		},
+	}
+
+	// Create logger
+	controllerLogger, err := common.NewLogger(&loggerConfig)
+	if err != nil {
+		_ = logFile.Close()
+		return fmt.Errorf("failed to create logger: %w", err)
+	}
+
+	// Create AWS clients using LocalStack container's configuration
+	awsConfig := harness.LocalStack.GetAWSClientConfig()
+
+	// Create DynamoDB client
+	dynamoClient, err := dynamodb.NewClient(awsConfig, controllerLogger)
+	if err != nil {
+		_ = logFile.Close()
+		return fmt.Errorf("failed to create dynamodb client: %w", err)
+	}
+
+	// Create metrics registry
+	metricsRegistry := prometheus.NewRegistry()
+
+	// Get available relays from config
+	availableRelays := make([]corev2.RelayKey, config.RelayCount)
+	for i := range config.RelayCount {
+		availableRelays[i] = corev2.RelayKey(i)
+	}
+
+	requestSigner, err := clients.NewDispersalRequestSigner(
+		ctx,
+		clients.DispersalRequestSignerConfig{
+			Region:   awsConfig.Region,
+			Endpoint: awsConfig.EndpointURL,
+			KeyID:    config.TestConfig.DisperserKMSKeyID,
+		})
+	if err != nil {
+		_ = logFile.Close()
+		return fmt.Errorf("failed to create dispersal request signer: %w", err)
+	}
+
+	// Build encoding manager configs
+	encodingManagerConfig := controller.DefaultEncodingManagerConfig()
+	encodingManagerConfig.NumRelayAssignment = uint16(config.RelayCount)
+	encodingManagerConfig.AvailableRelays = availableRelays
+	encodingManagerConfig.EncoderAddress = "localhost:34001"
+
+	// Build dispatcher configs
+	dispatcherConfig := controller.DefaultDispatcherConfig()
+	dispatcherConfig.FinalizationBlockDelay = 5
+	dispatcherConfig.BatchMetadataUpdatePeriod = 100 * time.Millisecond
+
+	// Chain state config
+	chainStateConfig := thegraph.DefaultConfig()
+	chainStateConfig.Endpoint = operatorStateSubgraphURL
+
+	// Start controller using shared function
+	controllerInstance, err := controller.StartController(
+		ctx,
+		controllerLogger,
+		ethClient,
+		dynamoClient,
+		config.MetadataTableNameV2,
+		metricsRegistry,
+		requestSigner,
+		gethcommon.HexToAddress(config.TestConfig.EigenDA.OperatorStateRetriever),
+		gethcommon.HexToAddress(config.TestConfig.EigenDA.ServiceManager),
+		gethcommon.HexToAddress(config.TestConfig.EigenDA.RegistryCoordinator),
+		operatorStateSubgraphURL,
+		encodingManagerConfig,
+		dispatcherConfig,
+		chainStateConfig,
+		// Optional components (not needed for tests)
+		nil,                                  // metricsServer
+		"",                                   // readinessProbePath
+		healthcheck.HeartbeatMonitorConfig{}, // No heartbeat monitor for tests
+		// Server config
+		// TODO(dmanc): Add server config
```

Comment:
Not sure about the below config. I ran `make run-payments-tests` and it seemed to pass without the below stuff?